dvc/dependency/repo.py

```diff
@@ -46,14 +48,86 @@ def _make_repo(self, **overrides):
         with external_repo(**merge(self.def_repo, overrides)) as repo:
             yield repo
 
+    def _get_checksum_in_repo(self, repo):
+        return repo.cache.local.get_checksum(
+            PathInfo(os.path.join(repo.root_dir, self.def_path))
+        )
+
+    # fileinfo is a dictionary containing "checksum" and "path"
+    def _get_fileinfo(self, updated=False):
+        rev_lock = None
+        if not updated:
+            rev_lock = self.def_repo.get(self.PARAM_REV_LOCK)
+
+        try:
+            with self._make_repo(rev_lock=rev_lock) as repo:
+                path = os.path.join(repo.root_dir, self.def_path)
+                try:
+                    checksum = repo.find_out_by_relpath(self.def_path).info[
+                        "md5"
+                    ]
+                except OutputNotFoundError:
+                    repo.state.load()
+                    checksum = repo.cache.local.get_checksum(PathInfo(path))
+
+                assert checksum
+
+                return {"checksum": checksum, "path": path, "repo": repo}
+
+        except NotDvcRepoError:
+            repo_path = cached_clone(
+                self.def_repo[self.PARAM_URL],
+                rev=rev_lock or self.def_repo.get(self.PARAM_REV),
+            )
+            path = os.path.join(repo_path, self.def_path)
+            return {"checksum": None, "path": path, "repo": None}
```

Comment:
You can toatlly `get_checksum()` a git file/dir, just use `self.repo.cache.local.get_checksum()`  instead of `repo.cache.local.get_checksum()`.

Reply:
However, this function may be called for this repo as well as with the "other repo" which has just been checked out and might not be DVC enabled.

Reply:
@fabiosantoscode It doesn't has to be dvc enabled, we are using our current repo's facilities to compute checksum for a file/dir on filesystem, no matter if it is in git or dvc repo itself.

Reply:
Ooooooh, nice :) I'll do just that!